test/src/tests-batchUploader.ts

```diff
@@ -329,6 +329,21 @@ describe('batch uploader', () => {
                 })
             });
 
+            it('should not throw an error when upload is called while storage has not been created yet', async () => {
+                window.localStorage.clear();
+                window.sessionStorage.clear();
+
+                window.mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = window.mParticle.getInstance();
+                const uploader = mpInstance._APIClient.uploader;
+
+                expect(uploader).to.equal(null)
+
+                // calling upload manually on an a null uploader should not result in an error
+                window.mParticle.upload();
```

Comment:
I think you should add an assertion to verify that an error actually doesn't throw.

IIRC the format should be something like:
`expect(() => { window.mParticle.upload()}).not.to.throw(new Error('<ERROR MESSAGE>'))`
or
`expect(() => { window.mParticle.upload()}).not.to.throw('<ERROR MESSAGE>')`

I think the gist is to make sure we're actually catching the right error message or to make sure throwing an error doesn't suspend other testing.

I would timebox this unless @rmi22186 feels strongly about it as well.

Reply:
yea - i actually was thinking the same thing, assuming that if an error threw, perhaps the test would still pass since there is no `expect`.  however, i did pull the code down yesterday to check and it throws an error and the test fails, so at least it's not a silent error.

here's a recommendation i found online as an example of how this would work, so try this, and if it doesn't work, I think we are good to go with what you currently have, and just add a comment above it that says something like


```
const { expect } = require('chai');
const sinon = require('sinon');

describe('MyFunction', () => {
    it('should not throw an error', () => {
        const myFunction = sinon.stub().returns(true); // Stub the function
        
        expect(() => myFunction()).to.not.throw();
    });
});
```

and if that doesn't work, then put a comment explaining why there is not `expect`
```
 // the following does not throw an error, but if it did, the test would fail because the error is not silent
or something like that. 
```

Reply:
I was able to get it working and not throw the error with a regex matching a part of the message while specifying the error type to be TypeError, tested on both before and after the fix and it worked as expected